src/lazy/binary/raw/v1_1/value.rs

```diff
@@ -246,6 +266,37 @@ impl<'top> LazyRawValue<'top, BinaryEncoding_1_1> for &'top LazyRawBinaryValue_1
 }
 
 impl<'top> LazyRawBinaryValue_1_1<'top> {
+    /// Constructs a lazy raw binary value from an input buffer slice that has been found to contain
+    /// a complete `FlexUInt`.
+    pub(crate) fn for_flex_uint(input: ImmutableBuffer<'top>) -> Self {
+        let encoded_value = EncodedValue {
+            encoding: ParameterEncoding::FlexUInt,
+            header: Header {
```

Comment:
Perhaps `Header` should actually be an `enum` so that it could have a `NoHeader` variant in addition to whatever other implementations you have going on?